api/src/main/java/org/openmrs/module/reporting/common/SqlRunner.java

```diff
@@ -144,6 +144,50 @@ public SqlResult executeSql(String sql, Map<String, Object> parameterValues) {
         return result;
 	}
 
+    /**
+     * Executes a Sql Script
+     */
+    public SqlIterator executeSqlToIterator(String sql) {
+
+        SqlIterator iterator = null;
+        log.info("Executing SQL...");
+
+        List<String> sqlStatements = new ArrayList<String>();
+        sqlStatements.addAll(parseSqlIntoStatements(sql));
+
+        Boolean originalAutoCommit = null;
+
+        try {
+            originalAutoCommit = connection.getAutoCommit();
+            connection.setAutoCommit(false);
+
+            for (String sqlStatement : sqlStatements) {
+                Statement statement = null;
+                try {
+                    statement = connection.createStatement();
+                    statement.setFetchSize(10);
```

Comment:
I'm not 100% sure how this is intended to work, but this looks problematic to me.  This seems to be adapted from the "SqlRunner.executeSql" method, which is designed to support a use case where one might be executing a SQL file that has multiple statements to execute sequentially, the last statement of which is a select statement to retrieve the ultimate result set.

The statements leading up to the last statement might build temporary tables, set variables, create stored procedures, and have all sorts of other logic in order to make the final query perform well and to simplify the overall query and add functionality.

So we can't really set the fetch size of every statement to 10 arbitrarily, as this will likely lead to some very unexpected results.

Perhaps an approach should be taken where all statements are executed fully until the last statement is reached.  And you only set the fetch size to 10 for that last statement, and this last statement forms the basis of the query that you are iterating on.  Does that make sense?